frontend/src/components/ChangeEmailScreen.js

```diff
@@ -1,75 +1,130 @@
 
 const config = require('../config');
 const BACKEND_URL = config.backend.uri;
 
+const useStyles = makeStyles((theme) => ({
+    form: {
+      //Input Field - Label Layout 
+      '& .MuiFormLabel-root': {
+          color: '#747474',
+        },
+        //Input Field - Border Layout 
+      '& .MuiOutlinedInput-root .MuiOutlinedInput-notchedOutline': {
+          border: '1.5px solid #747474'
+      }
+    }
+}));
+
+export default function AdminMenuItems (props) {
+    const classes = useStyles();
+    const [inputEmail, setPrimaryEmail] = useState("");
+    const [secondaryEmails, setSecondaryEmails] = useState([]);
+    const [inputError, setInputError] = useState(false);
+    const [open, setOpen] = useState(false);
+    const [errorMessage, setErrorMessage] = useState("");
 
-// calls email change route
-async function handleFormSubmit(email, setInputError){
-    // validate to make sure email is an email. We do this here because
-    // we don't want to overload the server with invalid emails
-    if (email && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(email)) {
-        setInputError(true);
-        return;
+    // catch enter rerendeing entire admin page
+    const handleKeyDown = (event) => {
+        if (event.key === 'Enter') {
+            handleFormSubmit(inputEmail, setInputError, setOpen)
+        }
     }
-    await fetch(`${BACKEND_URL}email/change`, {
+
+    useEffect(() => {
+        setSecondaryEmails(props.emails);
+    }, [props])
+
+    /**
+     * Updates the primary email in the database
+     * 
+     * @param {string} email - Primary email to update
+     * @param {function} setInputError - Error handling 
+     * @param {function} setOpen - Snackbox rendering 
+     */
+    async function handleFormSubmit(email, setInputError, setOpen) {
+        // make sure it is not a secondary email
+        if(secondaryEmails.includes(email)) {
+            setInputError(true);
+            setErrorMessage("This email is already listed as a secondary email.");  
+            return;
+        }
+        
+        // validate to make sure email is an email
+        if (email && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(email)) {
+            setInputError(true);
+            setErrorMessage("Enter a valid email address.");
+            return;
+        }
+
+        // change the primary email backend call
+        await fetch(`${BACKEND_URL}email/changePrimary`, {
             method: "POST",
             headers: {
                 "content-type": "application/json",
             },
             body: JSON.stringify({
-                "email": email
+                email: email
             })
         }).then(res => {
             if(res.ok){
-                alert("Email update successful!");
                 setInputError(false);
+                setOpen(true);
+                setPrimaryEmail("");
+                setErrorMessage("");
+                props.updatePrimaryEmail(email);
```

Comment:
I wasn't able to find a definition for updatePrimaryEmail in Email.js, and it's not passed through props. This throws an undefined error when I try to submit a new email.

Reply:
Forgot to remove that oops.. thanks